nodes/stampzilla-knx/main.go

```diff
@@ -35,44 +35,21 @@ func main() {
 
 		switch id[0] {
 		case "light":
-			config.Lock()
-			defer config.Unlock()
 
-			for _, light := range config.Lights {
-				if light.ID != id[1] {
-					continue
+			light := config.GetLight(id[1])
+			state.Bool("on", func(on bool) {
+				err := light.Switch(tunnel, on)
+				if err != nil {
+					logrus.Error()
 				}
-
-				for stateKey, newState := range state {
-					switch stateKey {
-					case "on":
-						diff, value, err := boolDiff(newState, device.State[stateKey])
```

Comment:
FYI man behöver inte kolla diffen för man kommer bara in i OnRequestStateChange med värden i state som har ändrats.